_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,138 @@
+---
+layout: post
+title: How to implement secure software development lifecycle
+author: Rinorragi
+excerpt: Sharing my experience and thoughts about secure software development lifecycle
+tags:
+- infosec
+- software security
+- devsecops
+---
+
+## What is secure software development lifecycle
+
+Have you ever found yourself wondering if the system you are implementing is secure enough? I have. Quite often actually. It is not an easy question to answer unless you are prepared. This blog post is about how to prepare yourself for that question. The short answer is Secure Software Development Lifecycle which I will call SSDLC from this point onwards. 
+
+Software development is usually divided into different phases. There can be more or there can be fewer depending on the software project management framework in use. Phases can exists in short iterations or in a one long waterfall or anything between. These phases could be for example:
+
+![ssdlc phases](/img/ssdlc/phases.png)
+
+SSDLC adds security requirements in each of the phases. This makes you consider security in all phases of the software project. 
+
+## Why does secure software development lifecycle matter
+
+It is time for mandatorial motivation! Why does this matter? Why I am reading this? Is this yet another SAFe or ITIL? 
+
+You should think SSDLC to be more like agile development. Where agile development advocates for adaptiveness and continual improvement the SSDLC should do the same. By itself it does not mean anything unless you are able to implement a useful set of tools and processes that supports your business goals. The same applies for SSDLC. Main goal is to provide a set of security oriented best practices for your existing development practices. 
+
+![processes](/img/ssdlc/processes.jpg)
+
+So what happened to "build the software and let security engineers test it afterwards"? The same happened as with all the other testing too. It is much cheaper to catch mistakes early. Security problems in requirements and/or in design are pretty expensive to fix if you let them go to production. You want to fix them early. To catch them early you need supporting routines to have a systematic approach in finding them. That is what SSDLC is all about. 
```

Comment:
Security problems in requirements and/or in design <- this needs to be rephrased